core/modules/replica/Configuration.cc

```diff
@@ -0,0 +1,343 @@
+        _xrootdHost                   (defaultXrootdHost),
+        _xrootdPort                   (defaultXrootdPort),
+        _xrootdTimeoutSec             (defaultXrootdTimeoutSec),
+        _workerTechnology             (defaultWorkerTechnology),
+        _workerNumProcessingThreads   (defaultWorkerNumProcessingThreads),
+        _workerNumFsProcessingThreads (defaultWorkerNumFsProcessingThreads),
+        _workerFsBufferSizeBytes      (defaultWorkerFsBufferSizeBytes),
+        _databaseTechnology           (defaultDatabaseTechnology),
+        _databaseHost                 (defaultDatabaseHost),
+        _databasePort                 (defaultDatabasePort),
+        _databaseUser                 (defaultDatabaseUser),
+        _databasePassword             (defaultDatabasePassword),
+        _databaseName                 (defaultDatabaseName),
+        _jobSchedulerIvalSec          (defaultJobSchedulerIvalSec) {
+}
+
+std::string Configuration::context() const {
+    static std::string const str = "CONFIG   ";
+    return str;
+}
+
+std::vector<std::string> Configuration::workers(bool isEnabled,
+                                                bool isReadOnly) const {
+    std::vector<std::string> names;
+    for (auto&& entry: _workerInfo) {
+        auto const& name = entry.first;
+        auto const& info = entry.second;
+        if (isEnabled) {
+            if (info.isEnabled and (isReadOnly == info.isReadOnly)) {
+                names.push_back(name);
+            }
+        } else {
+            if (not info.isEnabled) {
+                names.push_back(name);
+            }
+        }
+    }
+    return names;
+}
+
+std::vector<std::string> Configuration::databaseFamilies() const {
+
+    std::map<std::string, size_t> family2num;
+    for (auto&& elem: _databaseInfo) {
+        family2num[elem.second.family]++;
+    }
+    std::vector<std::string> families;
+    for (auto const& elem: family2num) {
+        families.push_back(elem.first);
+    }
+    return families;
+}
+
+bool Configuration::isKnownDatabaseFamily(std::string const& name) const {
+    return _replicationLevel.count(name);
+}
+
+size_t Configuration::replicationLevel(std::string const& family) const {
+    if (not _replicationLevel.count(family)) {
+        throw std::invalid_argument(
+                "Configuration::replicationLevel  unknown database family name: '" +
+                family + "'");
+    }
+    return _replicationLevel.at(family);
+}
+
+std::vector<std::string> Configuration::databases(std::string const& family) const {
+
+    if (not family.empty() and not _replicationLevel.count(family)) {
+        throw std::invalid_argument(
+                "Configuration::databases  unknown database family name: '" +
+                family + "'");
+    }
+    std::vector<std::string> names;
+    for (auto&& entry: _databaseInfo) {
+        if (not family.empty() and (family != entry.second.family)) { continue; }

```

Comment:
Is there such requirement in the LSST C++ Style Guide? :-)

Reply:
Not exactly, what we have there is:
https://developer.lsst.io/cpp/style.html#the-if-else-class-of-statements-should-have-the-following-form
https://developer.lsst.io/cpp/style.html#single-statement-if-else-for-or-while-statements-must-only-be-written-without-brackets-if-on-one-line

But why I don't like it is because is is very weird to see a block of code (`{}`) on a single line. Block implies that there is or will be more than one line of code in that block, and closing curly brace usually appears on a separate line of its own (except few cases like `} else {`).

Reply:
Agreed. Will get fixed.